snaps_k8s/provision/ansible_configuration.py

```diff
@@ -902,3 +929,189 @@ def delete_weave_interface(k8s_conf):
         }
         ansible_utils.apply_playbook(consts.K8_DELETE_WEAVE_INTERFACE,
                                      variables=pb_vars)
+def __get_loadbalancer_dict(config):
+    """
+    function to get loadbalancer dictionary
+    :param config:
+    :return:
+    """
+    logger.info("Argument List:\n config: %s", config)
+    for item in config_utils.get_ha_config(config):
+        return item.get(consts.HA_API_EXT_LB_KEY)
+                        
+
+def __get_hostname_map(hosts):
+    """Get hostname map function"""
+    logger.info("Argument List:\n hosts: %s", hosts)
+    hostname_map = {}
+    if hosts:
+        for i in range(len(hosts)):
+            hostname = hosts[i].get(consts.HOST_KEY).get('hostname')
+            host_ip = ""
+            ip_addr = hosts[i].get(consts.HOST_KEY).get(consts.IP_KEY)
+            if ip_addr:
+                host_ip = ip_addr
+            hostname_map[hostname] = host_ip
+
+    logger.info('Exit')
+    return hostname_map
+def __create_host_nodetype_map(hosts):
+    """Get Node types function"""
+    logger.info("Argument List:\n hosts: %s", hosts)
+    hostnode_map = {}
+    for host in hosts:
+        node_type = host.get(consts.HOST_KEY).get(consts.NODE_TYPE_KEY)
+        hostname = host.get(consts.HOST_KEY).get('hostname')
+        hostnode_map[hostname] = node_type
+    logger.info('Exit')
+    return hostnode_map
+    
+def __ha_configuration(loadbalancer_dict,pb_vars,k8s_conf,project_name):
+    """HA configuration """  
+    hosts = k8s_conf.get(consts.K8S_KEY).get(consts.NODE_CONF_KEY)
+    hostname_map = __get_hostname_map(hosts)
+    host_node_type_map = __create_host_nodetype_map(hosts)
+    logger.info('HA CONFIGURING')
+    ret_val_lb_config = __launch_ha_loadbalancer_conf(
+        hostname_map=hostname_map, host_node_type_map=host_node_type_map,
+        loadbalancer_dict=loadbalancer_dict, project_name=project_name,pb_vars=pb_vars)
+    ret_val_ha_config = __launch_kubespray_ha_configure(
+        hostname_map=hostname_map, host_node_type_map=host_node_type_map,
+        loadbalancer_dict=loadbalancer_dict, project_name=project_name,pb_vars=pb_vars)
+    if not(ret_val_lb_config and ret_val_ha_config):
+        logger.error('Alert !! Failed in HA Configuration')
+        exit(1)
```

Comment:
Please do not add explicit exit() calls. Raised exceptions should bubble all of the way up to main unless expected.